### PR TITLE
Allow building without groonga-httpd by setting WITHOUT_HTTPD env var

### LINKED ISSUE #146: Ability to install gem skipping groonga-httpd?
Hi,  I was wondering if it'd be possible to install the gem but passing a flag or option that would skip the compilation of `groonga-httpd`. I would like to use roonga in embedded mode so building Nginx unnecessary. Thanks!

### PR DESCRIPTION
**Pull request by tomas**
Fixes #146

### CODE REVIEW

File: ext/groonga/extconf.rb [hunk 1]
```diff
@@ -169,6 +169,7 @@ def configure_command_line(prefix)
   command_line << "--disable-document"
   command_line << "--disable-benchmark"
   command_line << "--without-cutter"
+  command_line << "--disable-groonga-httpd" if ENV['WITHOUT_HTTPD']
```
Comment (komainu8) [inline ext/groonga/extconf.rb]: I suggest that we add a prefix "RROONGA" for unifying the naming with other environmental variables.

File: ext/groonga/extconf.rb [hunk 2]
```diff
@@ -169,6 +169,7 @@ def configure_command_line(prefix)
   command_line << "--disable-document"
   command_line << "--disable-benchmark"
   command_line << "--without-cutter"
+  command_line << "--disable-groonga-httpd" unless ENV['RROONGA_WITH_HTTPD']
```
Comment (kou) [inline ext/groonga/extconf.rb]: Using `with_config` is better:

```ruby
command_line.concat(Shellwords.split(with_config("groonga-build-options") || ""))
```

```console
% gem install rroonga -- --with-groonga-build-options "--disable-groonga-httpd --other-option ..."
```

Comment (tomas) [reply ext/groonga/extconf.rb]: Would that replace all the above `command_line` stanzas?

Comment (kou) [reply ext/groonga/extconf.rb]: Groonga's `configure` overrides the same options by later options.
For example, `./configure --disable-document --enable-document` enables document.

Comment (tomas) [reply ext/groonga/extconf.rb]: Ok I think I'll just let you change the PR as see fit. :)

Comment (tomas) [reply ext/groonga/extconf.rb]: Any news?

Comment (kou) [reply ext/groonga/extconf.rb]: Ah, sorry. I misunderstood your comment.
I waited your change.

I'll update later.

Comment (tomas) [reply ext/groonga/extconf.rb]: Any news now? :)

Comment (kou) [reply ext/groonga/extconf.rb]: Now, I tried and I noticed that this branch is protected for this repository developers.
So I close this pull request and push to master directly.

**Pull request closed without merging**